Tests Fail on a Fresh Clone

I cloned the project and attempted to run the tests and ran into the following errors:

```
> fury-adapter-apib-serializer@0.1.2 test /Users/Peter/Code/localmed/fury-adapter-apib-serializer
> peasant test

🔎  Running tests...

   API Blueprint serializer adapter
     1) serializes basic.json
     2) serializes sample.json

   Executed 2 tests in 1 suites in 225ms
       0 passes
       2 failed

   1) API Blueprint serializer adapter serializes basic.json

       (/Users/Peter/Code/localmed/fury-adapter-apib-serializer/template.nunjucks) [Line 97, Column 20]
         TypeError: headers.include is not a function

       at Object.exports.prettifyError (node_modules/nunjucks/src/lib.js:34:15)

       33 |         var old = err;
       34 |         err = new Error(old.message);
       35 |         err.name = old.name;

       at node_modules/nunjucks/src/environment.js:486:31

       485 |                 if(err) {
       486 |                     err = lib.prettifyError(_this.path, _this.env.opts.dev, err);
       487 |                 }

       at new_cls.root [as rootRenderFunc] (eval at _compile (node_modules/nunjucks/src/environment.js:565:24), <anonymous>:611:3)
       at new_cls.render (node_modules/nunjucks/src/environment.js:479:15)

       479 |         _this.rootRenderFunc(
       480 |             _this.env,

       at node_modules/nunjucks/src/environment.js:312:35

       311 |             else {
       312 |                 syncResult = tmpl.render(ctx, cb);
       313 |             }

       at createTemplate (node_modules/nunjucks/src/environment.js:234:25)

       233 |                     if(cb) {
       234 |                         cb(null, tmpl);
       235 |                     }

       at handle (node_modules/nunjucks/src/environment.js:249:25)

       248 |                         src.loader = loader;
       249 |                         done(null, src);
       250 |                     }

       at node_modules/nunjucks/src/environment.js:263:21

       262 |                 else {
       263 |                     handle(null, loader.getSource(name));
       264 |                 }

       at next (node_modules/nunjucks/src/lib.js:207:13)

       206 |         if(i < arr.length) {
       207 |             iter(arr[i], i, next, cb);
       208 |         }

       at Object.exports.asyncIter (node_modules/nunjucks/src/lib.js:214:5)

       214 |     next();
       215 | };


   2) API Blueprint serializer adapter serializes sample.json

       (/Users/Peter/Code/localmed/fury-adapter-apib-serializer/template.nunjucks) [Line 51, Column 23]
         Error: Unable to call `item["key"]["toValue"]`, which is undefined or falsey

       at Object.exports.prettifyError (node_modules/nunjucks/src/lib.js:34:15)

       33 |         var old = err;
       34 |         err = new Error(old.message);
       35 |         err.name = old.name;

       at node_modules/nunjucks/src/environment.js:486:31

       485 |                 if(err) {
       486 |                     err = lib.prettifyError(_this.path, _this.env.opts.dev, err);
       487 |                 }

       at new_cls.root [as rootRenderFunc] (eval at _compile (node_modules/nunjucks/src/environment.js:565:24), <anonymous>:611:3)
       at new_cls.render (node_modules/nunjucks/src/environment.js:479:15)

       479 |         _this.rootRenderFunc(
       480 |             _this.env,

       at node_modules/nunjucks/src/environment.js:312:35

       311 |             else {
       312 |                 syncResult = tmpl.render(ctx, cb);
       313 |             }

       at new_cls.getTemplate (node_modules/nunjucks/src/environment.js:194:17)

       193 |             if(cb) {
       194 |                 cb(null, tmpl);
       195 |             }

       at new_cls.render (node_modules/nunjucks/src/environment.js:304:14)

       304 |         this.getTemplate(name, function(err, tmpl) {
       305 |             if(err && cb) {

       at Object.module.exports.render (node_modules/nunjucks/index.js:72:14)

       72 |     return e.render(name, ctx, cb);
       73 | };

       at Object.serialize (src/adapter.js:31:12)

       30 | export function serialize({api}, done) {
       31 |   nunjucks.render('template.nunjucks', {api}, (err, apib) => {
       32 |     if (err) {

       at Context.<anonymous> (test/adapter.js:34:15)

       34 |       adapter.serialize({api: refract}, (serializeErr, serialized) => {
       35 |         if (serializeErr) { return done(serializeErr); }


npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/Peter/.nvm/versions/node/v6.6.0/bin/node" "/Users/Peter/.nvm/versions/node/v6.6.0/bin/npm" "run" "test"
npm ERR! node v6.6.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! fury-adapter-apib-serializer@0.1.2 test: `peasant test`
npm ERR! Exit status 1
```

--- a/Changelog.md
+++ b/Changelog.md
@@ -1,3 +1,15 @@
+# 0.2.0 - 2017-01-24
+
+## Enhancements
+
+- Adds support for Fury ~> 2.3.
+
+## Bug Fixes
+
+- Fixes handling of parameter descriptions.
+- Fixes handling of MSON data structures and MSON attributes.
+
+
 # 0.1.2 - 2015-09-21
 
 - Fix handling of request/responses with empty bodies but including a description. These must make use of an empty body via `+ Body` or the description is considered the body content.

--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
   "devDependencies": {
     "chai": "^3.2.0",
     "coveralls": "^2.11.2",
-    "fury": "^1.0.3",
+    "fury": "^2.3.0",
     "glob": "^5.0.14",
     "peasant": "^0.5.2"
   },

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "fury-adapter-apib-serializer",
-  "version": "0.1.2",
+  "version": "0.2.0",
   "description": "API Blueprint serializer for Fury.js",
   "main": "./lib/adapter.js",
   "repository": {

--- a/src/adapter.js
+++ b/src/adapter.js
@@ -12,7 +12,8 @@ const env = nunjucks.configure(path.dirname(__dirname), {
   autoescape: false,
 });
 
-env.addFilter('mson', mson);
+env.addFilter('mson', mson.renderAttributes);
+env.addFilter('dataStructure', mson.renderDataStructure);
 env.addFilter('indent', indent);
 env.addFilter('bodyOnly', bodyOnly);
 env.addFilter('resourceShorthand', resourceShorthand);

--- a/src/mson.js
+++ b/src/mson.js
@@ -174,7 +174,7 @@ function handle(name, element, {parent = null, spaces = 4, marker = '+',
   }
 
   // Next, comes the optional example value
-  if (typeof element.content !== 'object') {
+  if (element.content && typeof element.content !== 'object') {
     if (parent && parent.element !== 'array') {
       str += ':';
     }
@@ -202,13 +202,23 @@ function handle(name, element, {parent = null, spaces = 4, marker = '+',
 /*
  * Render out a piece of MSON from refract element instances.
  */
-export default function render(mson) {
-  // Render *either* ### Title or + Attributes as the base element to support
-  // both a data structures section and resource / payload attributes.
-  const title = mson.title;
+export function renderDataStructure(dataStructure) {
+  const mson = dataStructure.content[0];
+  const title = mson.id;
 
-  return handle(title || 'Attributes', mson, {
-    initialMarker: title ? '###' : '+',
-    initialIndent: title ? false : true,
+  return handle(title, mson, {
+    initialMarker: '###',
+    initialIndent: false,
   });
 }
+
+export function renderAttributes(dataStructure) {
+  const mson = dataStructure.content[0];
+
+  return handle('Attributes', mson, {
+    initialMarker: '+',
+    initialIndent: true,
+  });
+}
+
+export default {renderDataStructure, renderAttributes};

--- a/template.nunjucks
+++ b/template.nunjucks
@@ -49,7 +49,7 @@ FORMAT: 1A
 
   {% for item in hrefVariables.content %}
     {% set typeAttributes = item.attributes.get('typeAttributes') %}
-    + {{ item.key.toValue() }}{% if item.value.toValue() %}: {{ item.value.toValue() }}{% endif %}{% if typeAttributes %} ({{ typeAttributes.toValue() }}){% endif %}{% if item.value.description %} - {{ item.value.description }}{% endif %}{% endfor %}
+    + {{ item.key.toValue() }}{% if item.value.toValue() %}: {{ item.value.toValue() }}{% endif %}{% if typeAttributes %} ({{ typeAttributes.toValue() }}){% endif %}{% if item.description %} - {{ item.description }}{% endif %}{% endfor %}
 
 {% endmacro %}
 

--- a/template.nunjucks
+++ b/template.nunjucks
@@ -126,7 +126,7 @@ FORMAT: 1A
 {{ renderParameters(resource.hrefVariables) }}
 {% endif %}
 {% if resource.dataStructure %}
-{{ resource.dataStructure.content|mson }}
+{{ resource.dataStructure|mson }}
 {% endif %}
 {% for transition in resource.transitions.content %}
   {{ renderTransition(transition, href) }}
@@ -151,7 +151,7 @@ FORMAT: 1A
 ## Data Structures
 
 {% for dataStructure in dataStructures.content %}
-{{ dataStructure.content|mson }}
+{{ dataStructure|dataStructure }}
 {% endfor %}
 {% endmacro %}
 

--- a/test/adapter.js
+++ b/test/adapter.js
@@ -21,19 +21,20 @@ describe('API Blueprint serializer adapter', () => {
     it(`serializes ${path.basename(file)}`, (done) => {
       let serializedRefract;
       let expectedBlueprint;
-      let refract;
+      let api;
 
       try {
         serializedRefract = require(file);
         expectedBlueprint = fs.readFileSync(apib, 'utf-8');
-        refract = fury.load(serializedRefract);
+        const parseResult = fury.load(serializedRefract);
+        api = parseResult.api;
       } catch (loadErr) {
         return done(loadErr);
       }
 
-      adapter.serialize({api: refract}, (serializeErr, serialized) => {
+      adapter.serialize({api}, (serializeErr, serialized) => {
         if (serializeErr) { return done(serializeErr); }
-        expect(expectedBlueprint.trim()).to.deep.equal(serialized.trim());
+        expect(serialized.trim()).to.deep.equal(expectedBlueprint.trim());
         done();
       });
     });

--- a/test/fixtures/basic.apib
+++ b/test/fixtures/basic.apib
@@ -2,7 +2,11 @@ FORMAT: 1A
 
 # My API
 
-#### GET /message
+## Group 
+
+### /message
+
+#### GET
 
 + Response 200 (text/plain)
 

--- a/test/fixtures/basic.json
+++ b/test/fixtures/basic.json
@@ -1,19 +1,119 @@
-[
-  "category", {"title": "My API"}, {}, [
-    ["resource", {}, {}, [
-      ["transition", {}, {}, [
-        ["httpTransaction", {}, {}, [
-          ["httpRequest", {}, {"method": "GET", "href": "/message"}, []],
-          ["httpResponse", {}, {"statusCode": 200, "headers": ["httpHeaders", {}, {}, [
-            ["member", {}, {}, {
-              "key": ["string", {}, {}, "Content-Type"],
-              "value": ["string", {}, {}, "text/plain"]
-            }]
-          ]]}, [
-            ["asset", {"classes": ["messageBody"]}, {}, "Hello, World!"]
-          ]]
-        ]]
-      ]]
-    ]]
+{
+  "element": "parseResult",
+  "content": [
+    {
+      "element": "category",
+      "meta": {
+        "classes": [
+          "api"
+        ],
+        "title": "My API"
+      },
+      "attributes": {
+        "meta": [
+          {
+            "element": "member",
+            "meta": {
+              "classes": [
+                "user"
+              ]
+            },
+            "content": {
+              "key": {
+                "element": "string",
+                "content": "FORMAT"
+              },
+              "value": {
+                "element": "string",
+                "content": "1A"
+              }
+            }
+          }
+        ]
+      },
+      "content": [
+        {
+          "element": "category",
+          "meta": {
+            "classes": [
+              "resourceGroup"
+            ],
+            "title": ""
+          },
+          "content": [
+            {
+              "element": "resource",
+              "meta": {
+                "title": ""
+              },
+              "attributes": {
+                "href": "/message"
+              },
+              "content": [
+                {
+                  "element": "transition",
+                  "meta": {
+                    "title": ""
+                  },
+                  "content": [
+                    {
+                      "element": "httpTransaction",
+                      "content": [
+                        {
+                          "element": "httpRequest",
+                          "attributes": {
+                            "method": "GET"
+                          },
+                          "content": []
+                        },
+                        {
+                          "element": "httpResponse",
+                          "attributes": {
+                            "statusCode": "200",
+                            "headers": {
+                              "element": "httpHeaders",
+                              "content": [
+                                {
+                                  "element": "member",
+                                  "content": {
+                                    "key": {
+                                      "element": "string",
+                                      "content": "Content-Type"
+                                    },
+                                    "value": {
+                                      "element": "string",
+                                      "content": "text/plain"
+                                    }
+                                  }
+                                }
+                              ]
+                            }
+                          },
+                          "content": [
+                            {
+                              "element": "asset",
+                              "meta": {
+                                "classes": [
+                                  "messageBody"
+                                ]
+                              },
+                              "attributes": {
+                                "contentType": "text/plain"
+                              },
+                              "content": "Hello, World!\n"
+                            }
+                          ]
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
   ]
-]
+}
+

--- a/test/fixtures/sample.apib
+++ b/test/fixtures/sample.apib
@@ -36,7 +36,7 @@ A frob does something.
 
 Gets information about a single frob instance
 
-+ Relation: Frob
++ Relation: self
 
 + Response 200 (application/json)
 

--- a/test/fixtures/sample.apib
+++ b/test/fixtures/sample.apib
@@ -23,19 +23,14 @@ A frob does something.
     + id (required)
     + tags (array)
         + (object)
-            Describes a tag.
+            + name (required) - Tag name.
+            + count: 23 (number)
+                I am some long content.
 
-            Each tag has a name and a number of times it was used.
+                - one
 
-            + Properties
-                + name (required) - Tag name.
-                + count: 23 (number)
-                    I am some long content.
-
-                    - one
-                    - two
-
-                    + Default: 1
+                - two
+                + Default: 1
 
 #### GET
 

--- a/test/fixtures/sample.json
+++ b/test/fixtures/sample.json
@@ -1,77 +1,376 @@
-[
-  "category", {"classes": ["api"], "title": "My API"}, {}, [
-    ["copy", {}, {}, "An API description."],
-    ["category", {"classes": ["resourceGroup"], "title": "My Group"}, {}, [
-      ["copy", {}, {}, "This is a group of resources"],
-      ["copy", {}, {"contentType": "text/plain"}, "Extra text"],
-      ["resource", {"title": "Frob"}, {
-        "href": "/frobs/{id}",
-        "hrefVariables": ["hrefVariables", {}, {}, [
-            ["member", {}, {"typeAttributes": ["required"]}, {
-              "key": ["string", {}, {}, "id"],
-              "value": ["string", {"description": "Unique identifier"}, {}, "123"]
-            }]
-          ]]
-        }, [
-        ["copy", {}, {}, "A frob does something."],
-        ["dataStructure", {}, {}, ["object", {}, {}, [
-          ["member", {}, {"typeAttributes": ["required"]}, {
-            "key": ["string", {}, {}, "id"],
-            "value": ["string", {}, {}, null]
-          }],
-          ["member", {}, {}, {
-            "key": ["string", {}, {}, "tags"],
-            "value": ["array", {}, {}, [
-                ["object", {"description": "Describes a tag.\n\nEach tag has a name and a number of times it was used."}, {}, [
-                  ["member", {"description": "Tag name."}, {"typeAttributes": ["required"]}, {
-                    "key": ["string", {}, {}, "name"],
-                    "value": ["string", {}, {}, null]
-                  }],
-                  ["member", {"description": "I am some long content.\n\n- one\n- two\n"}, {}, {
-                    "key": ["string", {}, {}, "count"],
-                    "value": ["number", {}, {"default": 1}, 23]
-                  }]
-                ]]
-            ]]
-          }]
-        ]]],
-        ["transition", {}, {"relation": "Frob"}, [
-          ["copy", {}, {}, "Gets information about a single frob instance"],
-          ["httpTransaction", {"title": "Get a frob"}, {}, [
-            ["httpRequest", {}, {"method": "GET"}, []],
-            ["httpResponse", {}, {"statusCode": 200, "headers": ["httpHeaders", {}, {}, [
-              ["member", {}, {}, {
-                "key": ["string", {}, {}, "Content-Type"],
-                "value": ["string", {}, {}, "application/json"]
-              }],
-              ["member", {}, {}, {
-                "key": ["string", {}, {}, "Authorization"],
-                "value": ["string", {}, {}, "Bearer abc123"]
-              }]
-            ]]}, [
-              ["asset", {"classes": ["messageBody"]}, {}, "{\n  \"id\": \"1\",\n  \"tags\": [\n    {\n      \"name\": \"foo\",\n      \"count\": 23\n    }\n  ]\n}\n"],
-              ["asset", {"classes": ["messageBodySchema"]}, {}, "{\"type\": \"object\",  \"properties\": {\"id\": {\"type\": \"string\"},\"tags\": {      \"type\": \"array\"}}}"]
-            ]]
-          ]]
-        ]],
-        ["transition", {"title": "Restart a frob"}, {"relation": "restart"}, [
-          ["copy", {}, {}, "Restart a frob instance"],
-          ["httpTransaction", {}, {}, [
-            ["httpRequest", {}, {"method": "POST"}, []],
-            ["httpResponse", {}, {"statusCode": 204}, [
-              ["copy", {}, {}, "There is no content."]
-            ]]
-          ]]
-        ]]
-      ]]
-    ]],
-    ["category", {"classes": ["dataStructures"]}, {}, [
-      ["dataStructure", {}, {}, ["object", {"title": "User"}, {}, [
-        ["member", {}, {"typeAttributes": ["required"]}, {
-          "key": ["string", {}, {}, "name"],
-          "value": ["string", {}, {}, null]
-        }]
-      ]]]
-    ]]
+{
+  "element": "parseResult",
+  "content": [
+    {
+      "element": "category",
+      "meta": {
+        "classes": [
+          "api"
+        ],
+        "title": "My API"
+      },
+      "attributes": {
+        "meta": [
+          {
+            "element": "member",
+            "meta": {
+              "classes": [
+                "user"
+              ]
+            },
+            "content": {
+              "key": {
+                "element": "string",
+                "content": "FORMAT"
+              },
+              "value": {
+                "element": "string",
+                "content": "1A"
+              }
+            }
+          }
+        ]
+      },
+      "content": [
+        {
+          "element": "copy",
+          "content": "An API description."
+        },
+        {
+          "element": "category",
+          "meta": {
+            "classes": [
+              "resourceGroup"
+            ],
+            "title": "My Group"
+          },
+          "content": [
+            {
+              "element": "copy",
+              "content": "This is a group of resources\n\nExtra text"
+            },
+            {
+              "element": "resource",
+              "meta": {
+                "title": "Frob"
+              },
+              "attributes": {
+                "href": "/frobs/{id}",
+                "hrefVariables": {
+                  "element": "hrefVariables",
+                  "content": [
+                    {
+                      "element": "member",
+                      "meta": {
+                        "description": "Unique identifier"
+                      },
+                      "attributes": {
+                        "typeAttributes": [
+                          "required"
+                        ]
+                      },
+                      "content": {
+                        "key": {
+                          "element": "string",
+                          "content": "id"
+                        },
+                        "value": {
+                          "element": "string",
+                          "content": "123"
+                        }
+                      }
+                    }
+                  ]
+                }
+              },
+              "content": [
+                {
+                  "element": "copy",
+                  "content": "A frob does something."
+                },
+                {
+                  "element": "dataStructure",
+                  "content": [
+                    {
+                      "element": "object",
+                      "meta": {
+                        "id": "Frob"
+                      },
+                      "content": [
+                        {
+                          "element": "member",
+                          "attributes": {
+                            "typeAttributes": [
+                              "required"
+                            ]
+                          },
+                          "content": {
+                            "key": {
+                              "element": "string",
+                              "content": "id"
+                            },
+                            "value": {
+                              "element": "string"
+                            }
+                          }
+                        },
+                        {
+                          "element": "member",
+                          "content": {
+                            "key": {
+                              "element": "string",
+                              "content": "tags"
+                            },
+                            "value": {
+                              "element": "array",
+                              "content": [
+                                {
+                                  "element": "object",
+                                  "content": [
+                                    {
+                                      "element": "member",
+                                      "meta": {
+                                        "description": "Tag name."
+                                      },
+                                      "attributes": {
+                                        "typeAttributes": [
+                                          "required"
+                                        ]
+                                      },
+                                      "content": {
+                                        "key": {
+                                          "element": "string",
+                                          "content": "name"
+                                        },
+                                        "value": {
+                                          "element": "string"
+                                        }
+                                      }
+                                    },
+                                    {
+                                      "element": "member",
+                                      "meta": {
+                                        "description": "I am some long content.\n\n- one\n\n- two"
+                                      },
+                                      "content": {
+                                        "key": {
+                                          "element": "string",
+                                          "content": "count"
+                                        },
+                                        "value": {
+                                          "element": "number",
+                                          "attributes": {
+                                            "default": 1
+                                          },
+                                          "content": 23
+                                        }
+                                      }
+                                    }
+                                  ]
+                                }
+                              ]
+                            }
+                          }
+                        }
+                      ]
+                    }
+                  ]
+                },
+                {
+                  "element": "transition",
+                  "meta": {
+                    "title": ""
+                  },
+                  "content": [
+                    {
+                      "element": "copy",
+                      "content": "Gets information about a single frob instance"
+                    },
+                    {
+                      "element": "httpTransaction",
+                      "content": [
+                        {
+                          "element": "httpRequest",
+                          "attributes": {
+                            "method": "GET"
+                          },
+                          "content": []
+                        },
+                        {
+                          "element": "httpResponse",
+                          "attributes": {
+                            "statusCode": "200",
+                            "headers": {
+                              "element": "httpHeaders",
+                              "content": [
+                                {
+                                  "element": "member",
+                                  "content": {
+                                    "key": {
+                                      "element": "string",
+                                      "content": "Content-Type"
+                                    },
+                                    "value": {
+                                      "element": "string",
+                                      "content": "application/json"
+                                    }
+                                  }
+                                },
+                                {
+                                  "element": "member",
+                                  "content": {
+                                    "key": {
+                                      "element": "string",
+                                      "content": "Authorization"
+                                    },
+                                    "value": {
+                                      "element": "string",
+                                      "content": "Bearer abc123"
+                                    }
+                                  }
+                                }
+                              ]
+                            }
+                          },
+                          "content": [
+                            {
+                              "element": "asset",
+                              "meta": {
+                                "classes": [
+                                  "messageBody"
+                                ]
+                              },
+                              "attributes": {
+                                "contentType": "application/json"
+                              },
+                              "content": "{\n  \"id\": \"1\",\n  \"tags\": [\n    {\n      \"name\": \"foo\",\n      \"count\": 23\n    }\n  ]\n}\n"
+                            },
+                            {
+                              "element": "asset",
+                              "meta": {
+                                "classes": [
+                                  "messageBodySchema"
+                                ]
+                              },
+                              "attributes": {
+                                "contentType": "application/schema+json"
+                              },
+                              "content": "{\n  \"type\": \"object\",\n  \"properties\": {\n    \"id\": {\n      \"type\": \"string\"\n    },\n    \"tags\": {\n      \"type\": \"array\"\n    }\n  }\n}\n"
+                            }
+                          ]
+                        }
+                      ]
+                    }
+                  ]
+                },
+                {
+                  "element": "transition",
+                  "meta": {
+                    "title": "Restart a frob"
+                  },
+                  "attributes": {
+                    "relation": "restart"
+                  },
+                  "content": [
+                    {
+                      "element": "copy",
+                      "content": "Restart a frob instance"
+                    },
+                    {
+                      "element": "httpTransaction",
+                      "content": [
+                        {
+                          "element": "httpRequest",
+                          "attributes": {
+                            "method": "POST"
+                          },
+                          "content": []
+                        },
+                        {
+                          "element": "httpResponse",
+                          "attributes": {
+                            "statusCode": "204"
+                          },
+                          "content": [
+                            {
+                              "element": "copy",
+                              "content": "There is no content."
+                            }
+                          ]
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+          ]
+        },
+        {
+          "element": "category",
+          "meta": {
+            "classes": [
+              "dataStructures"
+            ]
+          },
+          "content": [
+            {
+              "element": "dataStructure",
+              "content": [
+                {
+                  "element": "object",
+                  "meta": {
+                    "id": "User"
+                  },
+                  "content": [
+                    {
+                      "element": "member",
+                      "attributes": {
+                        "typeAttributes": [
+                          "required"
+                        ]
+                      },
+                      "content": {
+                        "key": {
+                          "element": "string",
+                          "content": "name"
+                        },
+                        "value": {
+                          "element": "string"
+                        }
+                      }
+                    }
+                  ]
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "element": "annotation",
+      "meta": {
+        "classes": [
+          "warning"
+        ]
+      },
+      "attributes": {
+        "code": 3,
+        "sourceMap": [
+          {
+            "element": "sourceMap",
+            "content": [
+              [
+                681,
+                15
+              ]
+            ]
+          }
+        ]
+      },
+      "content": "relation identifier contains illegal characters (only lower case letters, numbers, '-' and '.' allowed)"
+    }
   ]
-]
+}
+

--- a/test/fixtures/sample.json
+++ b/test/fixtures/sample.json
@@ -362,7 +362,7 @@
             "element": "sourceMap",
             "content": [
               [
-                681,
+                534,
                 15
               ]
             ]

--- a/test/fixtures/sample.json
+++ b/test/fixtures/sample.json
@@ -181,6 +181,9 @@
                   "meta": {
                     "title": ""
                   },
+                  "attributes": {
+                    "relation": "self"
+                  },
                   "content": [
                     {
                       "element": "copy",
@@ -347,29 +350,6 @@
           ]
         }
       ]
-    },
-    {
-      "element": "annotation",
-      "meta": {
-        "classes": [
-          "warning"
-        ]
-      },
-      "attributes": {
-        "code": 3,
-        "sourceMap": [
-          {
-            "element": "sourceMap",
-            "content": [
-              [
-                534,
-                15
-              ]
-            ]
-          }
-        ]
-      },
-      "content": "relation identifier contains illegal characters (only lower case letters, numbers, '-' and '.' allowed)"
     }
   ]
 }

